home-application/src/main/java/com/softserveinc/ita/homeproject/config/HomeUserDetailsService.java

```diff
@@ -0,0 +1,25 @@
+package com.softserveinc.ita.homeproject.config;
+
+import com.softserveinc.ita.homeproject.model.User;
+import com.softserveinc.ita.homeproject.repository.UserRepository;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class HomeUserDetailsService implements UserDetailsService {
+
+
+    private final UserRepository userRepository;
+
+    public HomeUserDetailsService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
```

Comment:
You've declared `throws UsernameNotFoundException`. Have you used it? Please be sure that you meet the contract.